Modules/_testcapi/dict.c

```diff
@@ -165,15 +252,21 @@ dict_getitemref(PyObject *self, PyObject *args)
     }
 }
 
+
+/*[clinic input]
+_testcapi.dict_getitemstringref
+    obj: object
+    value: object
+    attr_name: str(accept={robuffer}, zeroes=True)
```

Comment:
It looks like `str(accept={str, NoneType}, zeroes=True)` according to https://docs.python.org/3/howto/clinic.html?highlight=clinic#how-to-use-real-argument-clinic-converters-instead-of-legacy-converters

Reply:
Hi @erlend-aasland and @AlexWaygood  it doesn't work when we use `str(accept={str, NoneType}, zeroes=True)` grammar even if documentation support it. Would you like to take a look?

````
Error in file 'Modules/_testcapi/dict.c' on line 260:
str_converter: illegal combination of arguments (frozenset({<class 'NoneType'>, <class 'str'>}), False, True)
````

Reply:
The docs are inaccurate; the *accept* parameter should be `accept={robuffer, str, NoneType}`. See the definition in clinic.py:

https://github.com/python/cpython/blob/23a6db98f21cba3af69a921f01613bd5f602bf6d/Tools/clinic/clinic.py#L3927

Reply:
Also, it is possible to just use the legacy converter:

```suggestion
    attr_name: 'z#'
```